plugins/woocommerce/includes/class-wc-order.php

```diff
@@ -899,7 +928,15 @@ public function get_date_completed( $context = 'view' ) {
 	public function get_date_paid( $context = 'view' ) {
 		$date_paid = $this->get_prop( 'date_paid', $context );
 
-		if ( 'view' === $context && ! $date_paid && version_compare( $this->get_version( 'edit' ), '3.0', '<' ) && $this->has_status( apply_filters( 'woocommerce_payment_complete_order_status', $this->needs_processing() ? 'processing' : 'completed', $this->get_id(), $this ) ) ) {
+		/**
+		 * Filters the order status to set after payment complete.
+		 *
+		 * @param string   $status        Order status.
+		 * @param int      $order_id      Order ID.
+		 * @param WC_Order $this          Order object.
+		 */
+		$has_payment_complete_status = $this->has_status( apply_filters( 'woocommerce_payment_complete_order_status', $this->needs_processing() ? OrderStatus::PROCESSING : OrderStatus::COMPLETED, $this->get_id(), $this ) );
```

Comment:
We are better off moving this variable inside to keep filter invocation conditional as it was before (we'll get a nested if). Current placement leads to no-conditional filter invocation, which is a change of behaviour.

Reply:
Oh, ok. My thought was to move this to make the line shorter. I thought we would have no issues because the filter returns a list of statuses. Anyway, I just moved it back 👍